Umbraco Commerce v14 Docs Updates (#6307)
umbraco/UmbracoDocs: 14/umbraco-commerce/key-concepts/discount-rules-and-rewards.md

A basic label component is defined as follows:

```typescript
import { customElement, html, property } from "@umbraco-cms/backoffice/external/lit";
import { UmbLitElement } from "@umbraco-cms/backoffice/lit-element";

@customElement('my-discount-rule-label')
export class MyDiscountRuleLabelElement extends UmbLitElement {

    @property()
    value?:Record<string, unknown>;

    render() {
        return html`-- CREATE YOUR LABEL HERE --`
    }
}

export default MyDiscountRuleLabelElement;

declare global {
    interface HTMLElementTagNameMap {
        'my-discount-rule-label': MyDiscountRuleLabelElement;
    }
}

```

The component will pass a `Record<string, unknown>` value representing the rule/rewards configured values. Use this value to create your label.

Once defined, your component can be registered as a Property Editor UI via a manifest entry.

```javascript
const myDiscountRuleLabelManifest = {
    type: "propertyEditorUi",
    alias: "My.PropertyEditorUi.MyDiscountRuleLabel",
    name: "My Discount Rule Label",
    element: () => import('./my-discount-rule-label.element.js')
  };

  export const manifests = [ myDiscountRuleLabelManifest ];
```

{% hint style="info" %}

> Our component is registered as a Property Editor UI, but without a defined scheme, it won't display in Umbraco backoffice as a pickable property editor for use on Document Types.

Reviewer: ```suggestion
Without a defined scheme, the Property Editor UI will not display in Umbraco's backoffice as a pickable property editor for use on Document Types.
```
